src/mdns/mdns.cpp

```diff
@@ -159,6 +147,160 @@ void Publisher::OnHostResolved(const std::string &aHostName, const Publisher::Di
     }
 }
 
+Publisher::SubTypeList Publisher::SortSubTypeList(SubTypeList aSubTypeList)
+{
+    std::sort(aSubTypeList.begin(), aSubTypeList.end());
+    return aSubTypeList;
+}
+
+Publisher::TxtList Publisher::SortTxtList(TxtList aTxtList)
+{
+    std::sort(aTxtList.begin(), aTxtList.end(),
+              [](const TxtEntry &aLhs, const TxtEntry &aRhs) { return aLhs.mName < aRhs.mName; });
+    return aTxtList;
+}
+
+std::string Publisher::MakeFullServiceName(const std::string &aName, const std::string &aType)
+{
+    return aName + "." + aType + ".local";
+}
+
+std::string Publisher::MakeFullHostName(const std::string &aName)
+{
+    return aName + ".local";
+}
+
+void Publisher::AddServiceRegistration(ServiceRegistrationPtr aServiceReg)
+{
+    mServiceRegistrations[MakeFullServiceName(aServiceReg->mName, aServiceReg->mType)] = aServiceReg;
+}
+
+void Publisher::RemoveServiceRegistration(const std::string &aName, const std::string &aType)
+{
+    otbrLogInfo("Removing service %s.%s", aName.c_str(), aType.c_str());
+    mServiceRegistrations.erase(MakeFullServiceName(aName, aType));
+}
+
+Publisher::ServiceRegistrationPtr Publisher::FindServiceRegistration(const std::string &aName, const std::string &aType)
+{
+    return mServiceRegistrations[MakeFullServiceName(aName, aType)];
+}
```

Comment:
In all such `Find*` methods, Maybe we should add a `const` qualifier for this method. Also, we may use `find` to avoid creating dummy entries in the map.

Reply:
We are returning a non-const pointer to an element in `mServiceRegistrations`. Even if we can add a `const` qual, will it be misleading?